common4j/src/main/com/microsoft/identity/common/java/util/ApiResultUtil.kt

```diff
@@ -0,0 +1,63 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.util
+
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger
+import com.microsoft.identity.common.java.providers.nativeauth.IApiResponse
+import java.net.HttpURLConnection
+
+object ApiResultUtil {
+    /**
+     * Log IApiResponse objects. IResult objects are returned from all native auth endpoints
+     *
+     * @param tag    The log tag to use.
+     * @param response The result object to log.
+     */
+    // TODO: Do we consider response fields PII?
```

Comment:
Seems like we haven't decided on this yet @SammyO. Who do we reach out for to discuss? Privacy team?

Reply:
We only log the exposed fields, so i guess we can more so decide on this when looking at that PR

Reply:
I mean depends on what part of response you are logging. Tokens we should never log. username would be considered PII so we shouldn't log those either if they are part of response. Scopes should be fine to be logged. So you should review each field individually and decide that way.

Reply:
@SammyO

Reply:
@SaurabhMSFT  you can remove this TODO. As part of the next PRs we can review together whether API response fields are marked as loggable, i.e. whether they're PII.

Reply:
Removed the TODO